classes/file/ArticleFileManager.inc.php

```diff
@@ -509,6 +509,8 @@ function handleUpload($fileName, $fileStage, $fileId = null, $overwrite = false)
 
 		if (preg_match('/\.css$/', $_FILES[$fileName]['name'])) {
 			$articleFile->setFileType('text/css');
+		} else if (preg_match('/\.epub$/', $_FILES[$fileName]['name'])) {
+			$articleFile->setFileType('application/epub+zip');
 		} else {
 			$articleFile->setFileType($this->getUploadedFileType($fileName));
```

Comment:
As of 2.4.5, getUploadedFileType() now performs enhanced file extension checking which makes the above checks (existing text/css and new application/epub+zip) undesirable.

If MIME detection typically identifies an epub as application/zip, modify the String::mime_content_type() ambiguities to override: https://github.com/pkp/pkp-lib/blob/b0005fead5591a491e3034ef4debe3634da35ae3/classes/core/String.inc.php#L409

For extra credit, move these ambiguities to the database and create an interface for the Site Admin or Journal Manager to edit them.  I should have done that in the first place.